orchestrator/lib/coolfhir/contained.go

```diff
@@ -0,0 +1,98 @@
+package coolfhir
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/zorgbijjou/golang-fhir-models/fhir-models/fhir"
+)
+
+var (
+	ErrCareTeamNotFound = errors.New("CareTeam not found in FHIR Bundle")
+	ErrInvalidReference = errors.New("invalid reference, expecting local reference")
+)
+
+type ContainedResource struct {
+	Resource
+	Raw json.RawMessage `json:"-"`
+}
+
+func (r *ContainedResource) UnmarshalJSON(data []byte) error {
+	if err := json.Unmarshal(data, &r.Resource); err != nil {
+		return err
+	}
+
+	r.Raw = data
+
+	return nil
+}
+
+func (r *ContainedResource) MarshalJSON() ([]byte, error) {
+	return r.Raw, nil
+}
+
+func UpdateContainedResource(contained json.RawMessage, id *fhir.Reference, resourceType string, resource any) (json.RawMessage, error) {
+	if !IsLocalReference(id) {
+		return nil, ErrInvalidReference
+	}
+
+	data, err := json.Marshal(resource)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal data: %w", err)
+	}
+
+	var resources []ContainedResource
+
+	if err := json.Unmarshal(contained, &resources); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal contained resources: %w", err)
+	}
+
+	ref := *id.Reference
+
+	for i, containedResource := range resources {
+		if containedResource.ID == ref[1:] && containedResource.Type == resourceType {
```

Comment:
Tricky, do contained resources always follow the `#...` format? Maybe at least add a comment

I think we only actually support 1 CareTeam? Then you could also just find the contained resource to update by resource type.